internal/server/routes.go

```diff
@@ -56,10 +56,11 @@ func (s *Server) GenerateRoutes(promRegistry prometheus.Registerer) Routes {
 	apiGroup := router.Group("/",
 		metrics.Middleware(promRegistry),
 		DatabaseMiddleware(a.server.db), // must be after TimeoutMiddleware to time out db queries.
+		OrganizationFromDomain(s.options.Config.OrganizationName, s.options.Config.OrganizationDomain),
 	)
 	apiGroup.GET("/.well-known/jwks.json", a.wellKnownJWKsHandler)
 
-	authn := apiGroup.Group("/", authenticatedMiddleware())
+	authn := apiGroup.Group("/", authenticatedMiddleware(), OrganizationRequired())
```

Comment:
We will need to required orgs for some unauthenticated endpoints too (providers will be bound to orgs)

Reply:
It would be great to include this as a function call from `authenticatedMiddleware` and `unauthenticatedMiddleware`.  That reduces the number of things that depend on `gin.Context`.

Reply:
hmm. I don't really want to start combining middleware to save middleware functions.. We can still switch these to our own context easily enough